src/workspaces/features/RunPipelineDialog/RunPipelineDialog.tsx

```diff
@@ -109,25 +111,55 @@ const RunPipelineDialog = (props: RunPipelineDialogProps) => {
       </Alert>
     );
   }
+  const handleClose = () => {
+    onClose();
+    setShowVersionPicker(false);
+  };
 
   const parameters = form.formData.version?.parameters ?? [];
   return (
     <Dialog
       open={open}
-      onClose={onClose}
+      onClose={handleClose}
       maxWidth={parameters.length > 4 ? "max-w-3xl" : "max-w-2xl"}
     >
       <form onSubmit={form.handleSubmit}>
         <Dialog.Title>{t("Run pipeline")}</Dialog.Title>
         <Dialog.Content className="-m-1 max-h-[600px] overflow-y-auto p-1">
-          <Field name="version" label={t("Version")} required className="mb-6">
-            <PipelineVersionPicker
+          {!showVersionPicker ? (
+            <div className="mb-6 flex justify-start gap-x-1">
+              <p>
+                {t("This pipeline will run using the {{label}} ({{version}})", {
+                  label: "run" in props ? "same version" : "latest version",
```

Comment:
This will not necessary work for all languages, you need to have one line per use case.